include/RAJA/util/IndexLayout.hpp

```diff
@@ -0,0 +1,145 @@
+/*!
+ ******************************************************************************
+ *
+ * \file
+ *
+ * \brief   RAJA header file defining the IndexLayout class and IndexList classes.
+ *
+ ******************************************************************************
+ */
+
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+// Copyright (c) 2016-23, Lawrence Livermore National Security, LLC
+// and RAJA project contributors. See the RAJA/LICENSE file for details.
+//
+// SPDX-License-Identifier: (BSD-3-Clause)
+//~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~//
+
+#ifndef RAJA_INDEXLAYOUT_HPP
+#define RAJA_INDEXLAYOUT_HPP
+
+#include "RAJA/util/Layout.hpp"
+
+namespace RAJA 
+{
+
+template<typename IdxLin = Index_type>
+struct ConditionalIndexList {
+
+  RAJA_INLINE constexpr ConditionalIndexList(IdxLin* index_list_in) :
```

Comment:
It might be best to either remove the constructor and use aggregate initialization or make the data member private.

Reply:
Fixed.